tests/test_interrupted_sync.py

```diff
@@ -0,0 +1,198 @@
+                first_sync_count = first_sync_record_count.get(stream, 0)
+                second_sync_count = second_sync_record_count.get(stream, 0)
+                second_sync_messages = [record.get('data') for record in
+                                        second_sync_records.get(
+                                            stream, {}).get('messages', [])
+                                        if record.get('action') == 'upsert']
+                first_bookmark_value = first_sync_bookmarks.get('bookmarks', {stream: None}).get(stream)
+                second_bookmark_value = second_sync_bookmarks.get('bookmarks', {stream: None}).get(stream)
+                LOGGER.info(f"first_bookmark_value = {first_bookmark_value} \n second_bookmark_value = {second_bookmark_value}")
+                
+                if expected_replication_method == self.INCREMENTAL:
+                    
+                    replication_key = next(iter(expected_replication_keys[stream]))
+                    
+                    interrupted_bookmark_value = interrupted_sync_states['bookmarks'][stream]
+                    if stream in completed_streams:
+                        # Verify at least 1 record was replicated in the second sync
+                        self.assertGreaterEqual(second_sync_count,
+                                            1, 
+                                            msg="Incorrect bookmarking for {0}, at least one or more record should be replicated".format(stream))
+
+                    elif stream == interrupted_sync_states.get('currently_syncing', None):
+                        # For interrupted stream records sync count should be less equals
+                        self.assertLessEqual(second_sync_count,
+                                            first_sync_count,
+                                            msg="For interrupted stream - {0}, seconds sync record count should be lesser or equal to first sync".format(stream))
+
+                    elif stream in pending_streams:
+                        # First sync and second sync record count match
+                        self.assertGreaterEqual(second_sync_count,
+                                                first_sync_count,
+                                                msg="For pending sync streams - {0}, second sync record count should be more than or equal to first sync".format(stream))     
+
+                    else:
+                        raise Exception("Invalid state of stream {0} in interrupted state, please update appropriate state for the stream".format(stream))
+                    
+                    if stream in {'product_reviews', 'product_variants', 'order_fulfillments'}:
+                        repl_key = {'product_reviews': 'product_yotpo_id',
+                                    'order_fulfillments': 'order_id',
+                                    'product_variants': 'yotpo_product_id'
```

Comment:
These 3 streams product_reviews,product_variants, and  order_fulfillments are child streams of product and order streams, respectively. 
The bookmarking is maintained for each item( i.e. product id / order id) of the parent stream, while the DateTime value 'updated_at'/ 'created_at' is retrieved from the records (which is the actual replication key)
e.g. We are storing the state file for order_fulfillments then it will be a mapping of multiple order ids with their respective replication key values.
{order_id1: update_at1, order_id2: updated_at2, order_id3: updated_at3,.........,etc}

Reply:
Please use consistent formatting through out the code to define a dict variables.